lib/semantics/resolve-names.cc

```diff
@@ -3524,8 +3526,12 @@ bool DeclarationVisitor::Pre(const parser::SaveStmt &x) {
       auto kind{std::get<parser::SavedEntity::Kind>(y.t)};
       const auto &name{std::get<parser::Name>(y.t)};
       if (kind == parser::SavedEntity::Kind::Common) {
-        MakeCommonBlockSymbol(name);
-        AddSaveName(saveInfo_.commons, name.source);
+        if (currScope().kind() == Scope::Kind::Block) {
+          CheckNotInBlock("COMMON BLOCK NAME");  // C1107
```

Comment:
This message comes out as "COMMON BLOCK NAME statement is not allowed in a BLOCK construct". But unlike the other places `CheckNotInBlock` is used, "COMMON BLOCK NAME" is not a kind of statement.

I think this one needs a custom error message. Maybe add an overload of CheckNotInBlock that takes a `parser::MessageFixedText` to emit if the check fails.

Reply:
Also, this is C1108, not C1107.

Reply:
I agree with Tim on this, also note that the checks done in `CheckNotInBlock` is not really useful here because you have just tested it one line above: `if (currScope().kind() == Scope::Kind::Block)`. So you may want to simply emit a custom message here.

Reply:
I've created a new message for that error in  CheckSaveStmts. So we are not using CheckNotInBlock anymore for C1108.